### DISCUSSION TITLE
Gradient accumulation with distributed training

### DISCUSSION THREAD

**Discussion by IvanUkhov** [keras-team/keras#20578] [General]
Does anybody know if the implemented gradient accumulation can run under `tf.distribute.MirroredStrategy`?

**Comment by IvanUkhov** [comment]
Opened an issue: https://github.com/keras-team/keras/issues/20582.

**Comment by abhaskumarsinha** [comment]
If you are using tf backend, you can use gradient accumulation old TF way with it.

**Comment by IvanUkhov** [comment]
What is this old way?

**Comment by roebel** [comment]
I wanted to try this as well, but the problem currently appears to be that MirroredStrategy is broken in the latest versions of keras. see #21061

Besides that, I don't think there is an official old way.  That's why there were a few  wrappers in the wild as for example this one here: https://gradientaccumulator.readthedocs.io/en/latest/background/gradient_accumulation.html

**Comment by innat** [comment]
https://github.com/keras-team/tf-keras/issues/301
